sources/packages/backend/apps/api/src/route-controllers/student-appeal/_tests_/e2e/student-appeal.students.controller.submitStudentAppeal.e2e-spec.ts

```diff
@@ -0,0 +1,560 @@
+  });
+
+  it("Should determine partner's new income when the student submits an appeal for partner's new income.", async () => {
+    // Arrange
+    const application = await saveFakeApplication(appDataSource, {
+      student: student,
+    });
+    const endpoint = `/students/appeal/application/${application.id}`;
+    const payload = {
+      studentAppealRequests: [
+        {
+          formName: "partnerIncomeAppeal",
+          formData: {
+            partnerNewIncome: 6000,
+            applicationId: "",
+            applicationStatus: "",
+          },
+        },
+      ],
+    };
+
+    // Act
+    // Set application status to completed
+    application.applicationStatus = ApplicationStatus.Completed;
+    await applicationRepo.save(application);
+
+    // Mock form.io dry-run submission
+    formService.dryRunSubmission = jest.fn().mockResolvedValue({
+      valid: true,
+      formName: "partnerIncomeAppeal",
+      data: { data: payload.studentAppealRequests[0].formData },
+    });
+
+    // Assert
+    const response = await request(app.getHttpServer())
+      .post(endpoint)
+      .auth(studentToken, BEARER_AUTH_TYPE)
+      .send(payload)
+      .expect(HttpStatus.CREATED);
+
+    const newStudentAppealRequest = await studentAppealRequestRepo.findOne({
+      where: { studentAppeal: { id: response.body.id } },
+    });
+    const partnerNewIncome =
+      newStudentAppealRequest.submittedData.partnerNewIncome;
+    expect(partnerNewIncome).toBe(6000);
+
+    // Set the student appeal status to Approved to allow the next test case to be processed
+    newStudentAppealRequest.appealStatus = StudentAppealStatus.Approved;
+    await studentAppealRequestRepo.save(newStudentAppealRequest);
+  });
+
+  it("Should determine student's disability information when the student submits an appeal for disability information.", async () => {
+    // Arrange
+    const application = await saveFakeApplication(appDataSource, {
+      student: student,
+    });
+    const endpoint = `/students/appeal/application/${application.id}`;
+    const payload = {
+      studentAppealRequests: [
+        {
+          formName: "studentDisabilityAppeal",
+          formData: {
+            studentNewPDPPDStatus: "yes",
+          },
+        },
+      ],
+    };
+
+    // Act
+    // Set application status to completed
+    application.applicationStatus = ApplicationStatus.Completed;
+    await applicationRepo.save(application);
+
+    // Mock form.io dry-run submission
+    formService.dryRunSubmission = jest.fn().mockResolvedValue({
+      valid: true,
+      formName: "studentDisabilityAppeal",
+      data: { data: payload.studentAppealRequests[0].formData },
+    });
+
+    // Assert
+    const response = await request(app.getHttpServer())
+      .post(endpoint)
+      .auth(studentToken, BEARER_AUTH_TYPE)
+      .send(payload)
+      .expect(HttpStatus.CREATED);
+
+    const newStudentAppealRequest = await studentAppealRequestRepo.findOne({
```

Comment:
just a suggestion, expect newStudentAppealRequest to be defined before doing further checks.